src/xmipp/bindings/python/xmipp_base.py

```diff
@@ -1,93 +1,105 @@
-
+import re
+from collections import OrderedDict
+from distutils.spawn import find_executable
 
 from xmippLib import *
 import os
 import sys
+import subprocess
+
 
 def xmippExists(path):
     return FileName(path).exists()
 
 def getXmippPath(*paths):
-    ''' Return the path of the Xmipp installation folder
+    """ Return the path of the Xmipp installation folder
         if a subfolder is provided, will be concatenated to the path
-    '''
-    if 'XMIPP_HOME' in os.environ:
-        return os.path.join(os.environ['XMIPP_HOME'], *paths)
-    else:
-        # xmipp   =     build      <   bindings    <     python    <     xmipp_base.py
-        xmippHome = os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
-        if os.path.isfile(os.path.join(xmippHome, 'lib', 'libXmipp.so')):
-            print("Warning: XMIPP_HOME not found.\n"
-                  "Using an auto-detected directory: " + xmippHome)
+    """
+
+    candidates = []  # First candidate from XMIPP_HOME, second from this file path
+    envHome = os.environ.get('XMIPP_HOME', '')  # the join do nothing if second is absolute
+    candidates.append(os.path.join(os.environ.get('SCIPION_HOME', ''), envHome))
+    # xmipp    =          build      <   bindings    <     python    <     xmipp_base.py
+    candidates.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__)))))
+
+    for xmippHome in candidates:
+        if (os.path.isfile(os.path.join(xmippHome, 'lib', 'libXmipp.so')) and
+            os.path.isfile(os.path.join(xmippHome, 'bin', 'xmipp_mpi_reconstruct_significant'))):
             return os.path.join(xmippHome, *paths)
-        else:
-            raise Exception('Error: XMIPP_HOME environment variable not set')
 
-def getMatlabEnviron(*toolPaths):
-    """ Return an Environment prepared for launching Matlab
-    scripts using the Xmipp binding.
-    """
-    env = getEnviron()
-    env.set('PATH', os.environ['MATLAB_BINDIR'], Environ.BEGIN)
-    env.set('LD_LIBRARY_PATH', os.environ['MATLAB_LIBDIR'], Environ.BEGIN)
-    for toolpath in toolPaths:
-        env.set('MATLABPATH', toolpath, Environ.BEGIN)
-    env.set('MATLABPATH', os.path.join(os.environ['XMIPP_HOME'], 'libraries', 'bindings', 'matlab'),
-            Environ.BEGIN)
-    
-    return env
+    raise Exception("Error: Xmipp build directory not found. Searched at:\n - %s"
+                    % '\n - '.join(candidates))
+
+
+# def getMatlabEnviron(*toolPaths):
```

Comment:
Matlab is needed for the strain analysis. This function cannot be deleted

Reply:
I didn't find where it's used... In addition it contain errors and an equivalent that can be found in the xmipp3.Plugin() class. I agree, that one is used by protocol_volume_strain.py and then it's needed.

Reply:
@cossorzano, please, check if this function is needed or with those in the plugin is enough.